src/internal/serviceenv/service_env.go

```diff
@@ -281,6 +290,11 @@ func (env *NonblockingServiceEnv) initDBClient() error {
 }
 
 func (env *NonblockingServiceEnv) initListener() error {
+	// TODO: Change this to be based on whether a direct connection to postgres is available.
+	// A direct connection will not be available in the workers when the PG bouncer changes are in.
+	if env.Config().PPSPipelineName != "" {
+		return env.initProxyListener()
```

Comment:
for cleanliness, I'd rather have a `initProxyListener` and `initDirectListener` that return a `(col.PostgresListener, error)` and do the setting of it in one place here.